judgels-frontends/raphael/src/routes/uriel/contests/single/supervisors/ContestSupervisorAddForm/ContestSupervisorAddForm.tsx

```diff
@@ -1,33 +1,109 @@
-import { Button, Intent } from '@blueprintjs/core';
+import { Button, Intent, HTMLTable } from '@blueprintjs/core';
 import * as React from 'react';
 import { Field, InjectedFormProps, reduxForm } from 'redux-form';
 
 import { Required } from 'components/forms/validations';
 import { FormTextArea } from 'components/forms/FormTextArea/FormTextArea';
+import { FormTableInput } from 'components/forms/FormTableInput/FormTableInput';
+import { FormCheckbox } from 'components/forms/FormCheckbox/FormCheckbox';
+import { SupervisorManagementPermission } from 'modules/api/uriel/contestSupervisor';
 
 export interface ContestSupervisorAddFormData {
   usernames: string;
+  grantAllPermissions?: boolean;
+  grantedPermissions?: { [key: string]: boolean };
 }
 
 export interface ContestSupervisorAddFormProps extends InjectedFormProps<ContestSupervisorAddFormData> {
   renderFormComponents: (fields: JSX.Element, submitButton: JSX.Element) => JSX.Element;
 }
 
-const ContestSupervisorAddForm = (props: ContestSupervisorAddFormProps) => {
-  const usernamesField: any = {
-    name: 'usernames',
-    label: 'Usernames',
-    labelHelper: '(one username per line, max 100 users)',
-    rows: 20,
-    validate: [Required],
-    autoFocus: true,
+interface ContestSupervisorAddFormState {
+  grantAllPermissions: boolean;
+}
+
+class ContestSupervisorAddForm extends React.Component<ContestSupervisorAddFormProps, ContestSupervisorAddFormState> {
```

Comment:
Use initial values of all checkboxes to be checked (use redux-form's `initialValues` when instantiating this component; search the codebase for example)